lib/Hakyll/Core/Provider/Internal.hs

```diff
@@ -120,20 +118,20 @@ newProvider store ignore directory = do
   where
     oldKey = ["Hakyll.Core.Provider.Internal.newProvider", "oldFiles"]
 
-    -- Update modified if metadata is modified
-    maxmtime files = flip M.map files $ \rInfo@(ResourceInfo mtime meta) ->
-        let metaMod = fmap resourceInfoModified $ meta >>= flip M.lookup files
-        in rInfo {resourceInfoModified = maybe mtime (max mtime) metaMod}
+    -- Combine a resource with its metadata file
+    combine (ResourceInfo xTime xMeta) (ResourceInfo yTime yMeta) =
+        ResourceInfo (xTime `max` yTime) (xMeta <|> yMeta)
 
 
 --------------------------------------------------------------------------------
-getResourceInfo :: FilePath -> Set Identifier -> Identifier -> IO ResourceInfo
-getResourceInfo directory universe identifier = do
-    mtime <- fileModificationTime $ directory </> toFilePath identifier
-    return $ ResourceInfo (BinaryTime mtime) $
-        if mdRsc `S.member` universe then Just mdRsc else Nothing
-  where
-    mdRsc = fromFilePath $ flip addExtension "metadata" $ toFilePath identifier
+getResourceInfo :: FilePath -> Identifier -> IO (Identifier, ResourceInfo)
+getResourceInfo directory identifier = do
```

Comment:
Do we need to support asking for resource info about `.metadata` files?

Reply:
I guess I should rename this function to `getFileInfo`. It is called on every (non-ignored) file in the directory, and then it decides itself whether that file is metadata of another resource or a resource itself.